src/lib/hooks/swap/useSwapInfo.tsx

```diff
@@ -86,9 +88,15 @@ function useComputeSwapInfo(): SwapInfo {
     [trade.trade?.inputAmount, trade.trade?.outputAmount]
   )
 
-  // TODO(ianlapham): Fix swap slippage tolerance
-  // const allowedSlippage = useSwapSlippageTolerance(trade.trade ?? undefined)
-  const allowedSlippage = useMemo(() => new Percent(100), [])
+  // if auto slippage enabled, used computed tolerance based on trade, else, user cutsom user input
+  const slippageToleranceFromTrade = useSlippageToleranceFromTrade(trade.trade)
+  const userMaxSlippage = useAtomValue(maxSlippageAtom)
+  const allowedSlippage = useMemo(() => {
+    if (userMaxSlippage === 'auto') {
```

Comment:
make this logic a one-liner ternary: `() => userMaxSlippage === 'auto' ? slippageToleranceFromTrade : userMaxSlippage`